### PR TITLE
Added student type model and m2m field in Club

### PR DESCRIPTION
**Pull request by rohan-gupta8**
[ch2747]

Added new model to represent student type.

Model has m2m relationship with Club.

Added to populate to initialize data.

Returns student type only in Club view (not in Club List view to save bandwidth)

Frontend tasks pending

### CODE REVIEW

File: backend/clubs/migrations/0047_auto_20201023_1848.py [hunk 1]
```diff
@@ -1,15 +1,28 @@
-# Generated by Django 3.1.2 on 2020-10-18 21:41
+# Generated by Django 3.1.2 on 2020-10-23 22:48
```
Comment (ezwang) [inline backend/clubs/migrations/0047_auto_20201023_1848.py:1]: Since the `0047_profile_uuid_secret` migration is already in production, we may not want to delete this migration.

It is much easier to delete our local development db than to modify the production db. Try to avoid modifying any migrations committed to `master`, as those are already in production.

File: backend/clubs/models.py [hunk 2]
```diff
@@ -636,6 +637,14 @@ class Note(models.Model):
     created_at = models.DateTimeField(auto_now_add=True)
 
 
+class StudentType(models.Model):
+
```
Comment (ezwang) [inline backend/clubs/models.py:641]: A simple docstring here would be nice, to let people know what this is supposed to represent. For example:

```
Represents a student type that the club is intended for. For example, "International Students", "Transfer Students", or "Online Students".
```

File: backend/clubs/serializers.py [hunk 3]
```diff
@@ -745,6 +745,7 @@ class Meta:
             "founded",
             "image_url",
             "is_favorite",
+            "favorite_count",
```
Comment (ezwang) [inline backend/clubs/serializers.py:748]: Why was this line added? Not sure if good or bad, just curious.

Comment (rohan-gupta8) [reply backend/clubs/serializers.py:748]: accidentally deleted it

Comment (rohan-gupta8) [reply backend/clubs/serializers.py:748]: oh wait turns out it is redundant. I think rebasing might have fixed my deletion. will recommit without.

**Review by ezwang — APPROVED** [review]
LGTM!

**Review by esinx — APPROVED** [review]
This pr looks like a good guideline to adding new models :) looks very clear / good to me!
